sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -57,44 +84,114 @@ export default {
-        /* eslint-disable */
-        interval.value = setInterval(checkIdle, 30000);
-        /* eslint-enable */
+        interval.value = setInterval(checkIdle, 1000);
       }
     };
 
+    watch(
+      () => modalOpen.value,
+      (newValue: boolean) => {
+        if (newValue) {
+          countdownInterval.value = setInterval(() => {
+            if (countdown.value > 0) {
+              countdown.value = maximumIdleTime.value - idleTimeInSeconds.value;
+            }
+          }, 1000);
+        } else {
+          clearInterval(countdownInterval.value);
+          countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+        }
+      },
+    );
+
+    const extendUserSessionTime = () => {
+      setLastActivityTime();
+      resetIdleCheckerTimer();
+      modalOpen.value = false;
+      clearInterval(countdownInterval.value);
+      countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+    };
+
     const confirmExtendTimeModal = async () => {
+      modalOpen.value = true;
       if (await extendTimeModal.value.showModal()) {
-        lastActivityLogin.value = new Date();
-        resetIdleCheckerTimer();
+        extendUserSessionTime();
+      } else {
+        setLoggedOut();
       }
     };
 
     const checkIdle = () => {
-      const idleTimeInMintutes = getDatesDiff(
-        lastActivityLogin.value,
+      if (getLoggedOut()) {
+        logoff();
+      }
+      idleTimeInSeconds.value = getDatesDiff(
+        getLastActivityTime(),
         new Date(),
         "second",
-        true,
+        false,
       );
-      if (idleTimeInMintutes >= minimumIdleTime.value) {
+
+      // Exceeded user session time.
+      if (idleTimeInSeconds.value > maximumIdleTime.value) {
+        if (modalOpen.value === true) {
+          modalOpen.value = false;
+          clearInterval(countdownInterval.value);
+        }
+        // Logoff immediately in case session is expired.
+        logoff();
+      } else if (
+        idleTimeInSeconds.value >=
+        maximumIdleTime.value - COUNT_DOWN_TIMER_FOR_LOGOUT
+      ) {
         confirmExtendTimeModal();
+      } else {
+        // Close the modal in case idleTimeInSeconds has changed
+        // because lastAcitivityTime changed from activity in other tab/window.
+        modalOpen.value = false;
       }
     };
 
-    onMounted(async () => {
-      resetIdleCheckerTimer();
-    });
-
     const setLastActivityTime = () => {
       if (isAuthenticated.value) {
-        lastActivityLogin.value = new Date();
+        localStorage.setItem(
+          LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+          new Date().getTime().toString(),
+        );
       }
     };
+
+    const getLastActivityTime = () => {
+      const localStorageValue = localStorage.getItem(
+        LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+      );
+      if (localStorageValue) {
+        return new Date(+localStorageValue);
+      }
+      return new Date();
+    };
+
+    const setLoggedOut = () => {
```

Comment:
I am not able to follow this exactly. If the time is shared in local storage and both tabs/sessions would be monitoring the same local storage, why do we need this extra variable in the storage?

Reply:
last activity time is share among the tabs but how to signal to the tabs that the user clicked on "NO" for the time extension?   "No" in this case means "Log me out now". Then the other will detect the user was logged out and  log themselves out too.

Reply:
I see the point. For cross-tab communication, it is possible also to use storage events, as below. I never used it but seems straightforward. For the mentioned scenario maybe an event would be better.
```ts
window.addEventListener('storage', (event) => {
 if (event.storageArea != localStorage) return;
 if (event.key === 'loggedIn') {
   // Do something with event.newValue
 }
});
```

Reply:
ok. I see. I'm just not sure how compatible it is across the modern browsers.

Reply:
Jus adding here for history, as per this page it is pretty compatible: https://www.w3schools.com/jsref/event_storage_url.asp
![image](https://user-images.githubusercontent.com/61259237/207665289-96f6ca97-a544-48ef-a99d-8adc18e302a9.png)